### DISCUSSION TITLE
TP2 Ejercicio 7 - Potenciación

### DISCUSSION THREAD

**Discussion by NicoHermosilla** [INGCOM-UNRN-P1/cursada-2023#75] [Preguntas y respuestas] [answered]
estoy teniendo problemas para plantear la potenciacion con base negativa, el codigo que hice sirve para hacer con bases positivas y cuando el exponente es 0. No estaria sabiendo como hacer para ese tipo de caso.
https://github.com/INGCOM-UNRN-P1/tp2-2023-NicoHermosilla

**Comment by martinvilu** [comment]
Está perfecto con la parte positiva, aunque posible, la parte negativa es bastante más compleja.

**Reply by NicoHermosilla** [reply]
Logre resolver utilizando la funcion signo hecha en el tp1,  es valido eso?
https://github.com/INGCOM-UNRN-P1/tp2-2023-NicoHermosilla
comparto el link del repo para que vea como quedo.

**Reply by martinvilu** [reply, marked as answer]
>  es valido eso?

![legal-star](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/56625/f0a754f7-360a-4127-9ec5-6ef88f4f64c4)

Ocvio y mirá esto

```C
   if(signo(numero) == -1) 
   {
      numero = -numero;
   }

```
Lo podes reemplazar por, 
```C
numero = numero * signo(numero);
```
Si es `1`, el signo no cambia, si es `-1`, lo invierte, y si es `0`, va a seguir siéndolo.

Lo único que veo es que faltan comentarios de documentación.

**Reply by NicoHermosilla** [reply]
Genial, sobre los comentarios los voy a agregar cuando todos los ejercicios resueltos. Gracias!!

**Reply by martinvilu** [reply]
Es una buena idea arrancar con los comentarios, te pueden ayudar a no desviarte en lo que estas implementando.
